tests/handlers/test_oidc.py

```diff
@@ -591,6 +598,63 @@ def test_exchange_code(self):
             yield defer.ensureDeferred(self.handler._exchange_code(code))
         self.assertEqual(exc.exception.error, "some_error")
 
+    @defer.inlineCallbacks
```

Comment:
adding new async test methods isn't great, though I realise you're trying to reuse existing code.